src/java.desktop/share/classes/javax/swing/plaf/metal/MetalRootPaneUI.java

```diff
@@ -536,7 +536,7 @@ public Dimension minimumLayoutSize(Container parent) {
             }
 
             return new Dimension(Math.max(Math.max(cpWidth, mbWidth), tpWidth) + i.left + i.right,
-                                 cpHeight + mbHeight + tpWidth + i.top + i.bottom);
+                                 cpHeight + mbHeight + tpHeight + i.top + i.bottom);
```

Comment:
Is it possible to check this change by some testcase?

Reply:
I tried at the time of PR but unable to check as MetalRootLayout is private so not able to access its methods. That is why I ran all client-tier3 jtreg and jck tests, and mach5 job link is posted in JBS.

Reply:
If the MetalRootLayout is private and it is not possible to trigger this code in other ways, then it means we can delete it?

Reply:
MetalRootLayout class and methods are private but it is called from MetalRootPaneUI so we cannot delete it 
but I could not find any ways to invoke MetalRootLayout methods from testcase.

Reply:
The MetalRootPaneUI installs the MetalRootLayout as a layout manager for the current JRootPane. And it should be used from the Container.preferredSize()

Reply:
I tried following testcase snippet
               ```
 JRootPane r = new JRootPane();
                JFrame f = new JFrame();
                f.getRootPane().setUI(new MetalRootPaneUI());
                f.getRootPane().setWindowDecorationStyle(JRootPane.FRAME);
                f.getRootPane().getUI().installUI(r);
                r.getContentPane().add(new JLabel("foo"));
                System.out.println("Preferred Size: " + r.getPreferredSize());
```
but it uses JRootPane.RootLayout layout manager and not MetalRootLayout so it calls Rootlayout.preferredLayoutSize and not MetalRootLayout.preferredlayoutSize

Reply:
You need to set decorations for the root pane:
```
        UIManager.setLookAndFeel(new MetalLookAndFeel());
        JFrame frame = new JFrame();
        frame.getRootPane().setWindowDecorationStyle(JRootPane.FRAME);
        frame.pack();
        System.out.println("Layout: " + frame.getRootPane().getLayout());
========
Layout: javax.swing.plaf.metal.MetalRootPaneUI$MetalRootLayout@69d9c55
```

Reply:
I tried with 
```
JFrame frame = new JFrame();
                frame.setSize(new Dimension(500, 800));
                frame.getRootPane().setWindowDecorationStyle(JRootPane.FRAME);
                //frame.pack();
                frame.setVisible(true);
                System.out.println("Layout: " + frame.getRootPane().getLayout());
                System.out.println("PreferredSize " + frame.getRootPane().getPreferredSize());
```
but it seems tpWidth and tpHeight what is being changed is the width and height of MetalTitlePane which seems always comes as 23 for both no matter what frame size is set, so I guess using any variable will do but logically, it will be good to use "tpHeight" for height calculation.
If it is believed to continue with same logical but not technical error, I will close this PR without integrating.

Reply:
The bug submitter said that it is somehow affected his application. So I assume there is a way to trigger this bug in the MetalRootLayout.

Reply:
From MetalTitlePane.java, the same "height" is being used for width and height so the value is same for both.
```
private class TitlePaneLayout implements LayoutManager {
        public void addLayoutComponent(String name, Component c) {}
        public void removeLayoutComponent(Component c) {}
        public Dimension preferredLayoutSize(Container c)  {
            int height = computeHeight();
            return new Dimension(height, height);
        }
```

Reply:
There is no reproducible testcase and the submitter id is not active anymore so cannot ask him and I am not sure of any other way, so if this is not accepted without a reproducing testcase, I will close this PR.